src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java

```diff
@@ -342,6 +342,12 @@ private void startNextDownloaders() throws SnowflakeSQLException {
             nextChunkToConsume,
             getPrefetchMemRetry);
         currentMemoryUsage.addAndGet(-neededChunkMemory);
+        nextChunk.getLock().lock();
+        try {
+          nextChunk.setDownloadState(DownloadState.FAILURE);
```

Comment:
If you set it to a failure state, how will this data be 'forcefully' retrieved later on?

Reply:
The logic is here: https://github.com/snowflakedb/snowflake-jdbc/pull/690/files#diff-7167db85e495a75470b94425d2f222db53b0854a4a06c745672814b1d3d598ecL630 
When the chunkStatus is not success, it will ignore memory limit and submit a download job.

Reply:
previous link does not work. logic here https://github.com/snowflakedb/snowflake-jdbc/blob/5a9f34c01283731f3b3f98e58a64ae6380f1f062/src/main/java/net/snowflake/client/jdbc/SnowflakeChunkDownloader.java#L642